src/pybind/mgr/nfs/export.py

```diff
@@ -463,13 +463,26 @@ def _create_user_key(
             self.rados_pool, cluster_id, fs_name)
         access_type = 'r' if fs_ro else 'rw'
 
-        ret, out, err = self.mgr.check_mon_command({
+        ret, out, err = self.mgr.mon_command({
             'prefix': 'auth get-or-create',
             'entity': 'client.{}'.format(entity),
             'caps': ['mon', 'allow r', 'osd', osd_cap, 'mds', 'allow {} path={}'.format(
                 access_type, path)],
             'format': 'json',
         })
+        if ret == -errno.EINVAL and 'does not match' in err:
+            ret, out, err = self.mgr.check_mon_command({
+                'prefix': 'auth caps',
+                'entity': 'client.{}'.format(entity),
+                'caps': ['mon', 'allow r', 'osd', osd_cap, 'mds', 'allow {} path={}'.format(
+                    access_type, path)],
+                'format': 'json',
+            })
+            ret, out, err = self.mgr.check_mon_command({
+                'prefix': 'auth get',
+                'entity': 'client.{}'.format(entity),
+                'format': 'json',
+            })
```

Comment:
Raise exception in case of other errors

Reply:
check_mon_command raises already

Reply:
I was talking about the `mon_command()` above

https://github.com/ceph/ceph/pull/42073/files#diff-d00d146481b43490d2e6e79fcf061bf8fbd7364614e01613320d6e2234025c02R471-R476